### PR TITLE
feat: add classes to purge as json to a separate file and rename components.css to warp.css

### PR DESCRIPTION
**Pull request by AnnaRybkina**
A part of https://nmp-jira.atlassian.net/jira/software/projects/WARP/boards/15?label=web&selectedIssue=WARP-302

This PR should be tested together with https://github.com/warp-ds/drive/pull/160 and test instructions are provided there.

What has been done:
1. Publish css repo to github pages and with that publish `classesToPurge.json` https://warp-ds.github.io/css/component-classes/classesToPurge.json. We should ofc change the branch name before we merge this one(to next)
2. Add pre-commit hook to generate new types if applicable for the classes and update json file with new classes
3. remove duplicates from the classes list

~**I see comments about publishing classes to EIK to sync versions in other PR:s. It is a good suggestion to sync the versions. If it is better let’s publish the list to EIK instead of gh pages. Please comment in this PR**~

**After the merge of this one we can adjust the url from where we fetch the classes in the corresponding drive PR. And remove GH pages publishing for this repo**

UPDATE: after a sync with the rest of the team we have decided following:
- rename `components.css` to `warp.css`
- and because of 👆 we release a major version of the css

### CODE REVIEW

File: component-classes/vite.config.js [hunk 1]
```diff
@@ -15,6 +16,14 @@ export default () => ({
       insertTypesEntry: true,
       outputDir: '../dist',
     }),
+    {
+      name: 'build-script',
+      buildStart() {
```
Comment (AnnaRybkina) [inline component-classes/vite.config.js:21]: This can also be added to the build step as script. If we want to keep it in the same place

**Comment by BalbinaK** [comment]
> I see comments about publishing classes to EIK to sync versions in other PR:s. It is a good suggestion to sync the versions. If it is better let’s publish the list to EIK instead of gh pages. Please comment in this PR

It may be related to my comment [here](https://github.com/warp-ds/drive/pull/160/files#r1352289863) but if not, then I just want to confirm that I agree that publishing `classesToPurge.json` to Eik (but not to NPM) would help us ensure synchronisation between assets published from this package 😊

**Comment by AnnaRybkina** [comment]
> > I see comments about publishing classes to EIK to sync versions in other PR:s. It is a good suggestion to sync the versions. If it is better let’s publish the list to EIK instead of gh pages. Please comment in this PR
> 
> It may be related to my comment [here](https://github.com/warp-ds/drive/pull/160/files#r1352289863) but if not, then I just want to confirm that I agree that publishing `classesToPurge.json` to Eik (but not to NPM) would help us ensure synchronisation between assets published from this package 😊

https://github.com/warp-ds/css/pull/56/commits/fd1586dcaaa0e2f3a4ee52e8239ecfe3a5160200 added json file to dist, it will be published to Eik

File: component-classes/scripts/updateClassesToPurge.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import fs from 'node:fs';
+import { classes } from '../classes.js';
+
+export default () => {
+  fs.writeFileSync('./classesToPurge.json', JSON.stringify(classes), 'utf-8');
```
Comment (BalbinaK) [inline component-classes/scripts/updateClassesToPurge.js:5]: Do we have use for this file? Can't find any mentions of `./component-classes/classesToPurge.json` in this repo.

File: component-classes/vite.config.js (hunk 1 above)
Comment (BalbinaK) [reply component-classes/vite.config.js:21]: Do you mean to have the `build:cc` command look something like this instead?

```
    "build:cc": "cd component-classes && rm -rf *.d.ts && tsc && node ./scripts/updateClassesToPurge.js"
``` 

If so, that does sound like a good idea to me. Then we'd show everything that's happening in the `build:cc` script in one place.

File: component-classes/scripts/updateClassesToPurge.js (hunk 2 above)
Comment (AnnaRybkina) [reply component-classes/scripts/updateClassesToPurge.js:5]: We publish this file to Eik/gh pages

Comment (BalbinaK) [reply component-classes/scripts/updateClassesToPurge.js:5]: I thought we are publishing the one that is saved in the next line, under `../dist/classesToPurge.json`.

Comment (BalbinaK) [reply component-classes/scripts/updateClassesToPurge.js:5]: Also, do we still want to publish this to GH pages?

**Review by imprashast — APPROVED** [review]

File: component-classes/scripts/updateClassesToPurge.js (hunk 2 above)
Comment (AnnaRybkina) [reply component-classes/scripts/updateClassesToPurge.js:5]: I will remove publishing after this PR is merged. No we don't need it anymore

Comment (AnnaRybkina) [reply component-classes/scripts/updateClassesToPurge.js]: Refactored here. I thought that we can commit the file but yea, maybe it is needed. https://github.com/warp-ds/css/pull/56/commits/ec090b2ae41b4340c806f380800c763fad5932b1

**Review by BalbinaK — APPROVED** [review]
🎉

**Pull request merged**